podcast/2.18.md

```diff
@@ -0,0 +1,545 @@
+
+**Kelly Norton:**
+
+Yeah. I think this really becomes acute the moment you introduce client libraries or protobuffs into your ... So anytime you start defining the interfaces between things, boy, does it become acute really quickly. So I always have a joke that anytime the question of, "Hey, we're going to start doing services and we use a GitHub model." And I go, "But where are you going to store your protobuff definitions?" And inevitably, everyone laughs because there has been a long, drawn-out argument about where the protobuff definitions live. And I mean, my basic pitch for a Monorepo is it's actually really hard to anticipate where the complexities are going to come in the future in terms of where code lives and what it takes to change something in one place and get it to the other place.
+
+Monorepo is like, I mean, they basically solved that problem because everything's in the same place. And I think people ... I mean the argument I had heard against them after leaving Google was that, "Well the Google model really requires a lot of tooling." And originally, I was sympathetic to that. And I go, "Yeah, it did require a lot of tooling." And then after doing the non-Google model for a little while, I realized like, "Wow, we have a lot of tooling to support this sort of multi-repo world." So would really just rather have the simpler mental model about how to make code changes and just invest in the tooling because I invested in the tooling anyway; we still built a ton of tooling.
+
+To give you a concrete example inside of Mailchimp: we have some PHP libraries and we distribute them via Composer internally. So, sadly, even though that fortunately Composer exists – and Composer is kind of like the package manager for PHP, and it's good; it does its job well – it's really intended for pulling in open-source projects, but it does the job. But it just adds so much additional complexity to our changes to our shared libraries because we have to make a change to the library. And then we actually have automations that then push pull requests to all the various apps that use that library.
+
+And again, this is all tooling that was required to support this. And somebody has to maintain that tooling. And it's not like it makes things easier because we use a continuous deployment model. So if your change to that shared library doesn't land well in the main app, the best course of action is to revert it out all the way back to the library so that you're not blocking deploys for other people, which is a lot of work. I mean, it really complicates some day-to-day flows. So I continue to ... The more I use the non-Monorepo world, or the more I'm sort of in the non-Monorepo world, the more I love the Monorepo world.
```

Comment:
```suggestion
And again, this is all tooling that was required to support this. And somebody has to maintain that tooling. And it's not like it makes things easier because we use a continuous deployment model. So if your change to that shared library doesn't land well in the main app, the best course of action is to revert it out all the way back to the library so that you're not blocking deploys for other people, which is a lot of work. I mean, it really complicates some day-to-day flows. So I continue to ... The more I use the non-monorepo world, or the more I'm sort of in the non-monorepo world, the more I love the monorepo world.
```